lib/fs/basicfs.go

```diff
@@ -52,21 +79,36 @@ func newBasicFilesystem(root string) *BasicFilesystem {
 		}
 	}
 
-	// Attempt to enable long filename support on Windows. We may still not
-	// have an absolute path here if the previous steps failed.
-	if runtime.GOOS == "windows" {
-		if filepath.IsAbs(root) && !strings.HasPrefix(root, `\\`) {
-			root = `\\?\` + root
-		}
-		// If we're not on Windows, we want the path to end with a slash to
-		// penetrate symlinks. On Windows, paths must not end with a slash.
-	} else if root[len(root)-1] != filepath.Separator {
-		root = root + string(filepath.Separator)
+	return root
+}
+
+func checkRootSymlink(root string) (string, error) {
+	// Check if root is a symlink
+	info, infoErr := underlyingLstat(root)
+	if infoErr != nil {
+		return root, infoErr
+	}
+	if info.Mode()&os.ModeSymlink == 0 {
+		return root, errors.New("Not a symlink")
```

Comment:
As later on we do not actually act on an error here, why do you do this symlink check (and later on the directory check)?

Reply:
It's probably safe to skip this check, using error on `EvalSymlink` as indication that this isn't a symlink instead. The latter directory check is also likely unnecessary since if it is a symlink and not a directory, than it will return an error later in path checking. I'll go ahead and slim this code.